Foundation/NSStream.swift

```diff
@@ -209,20 +227,32 @@ public class NSOutputStream : Stream {
         return self.init(toMemory: ())
     }
     
-    public override func propertyForKey(_ key: String) -> AnyObject? {
-        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    public override var streamError: NSError?{
+        let error = CFWriteStreamCopyError(_stream)
+        return error?._nsObject
     }
     
     public  override func setProperty(_ property: AnyObject?, forKey key: String) -> Bool {
+        guard let property = property else { return false }
         return CFWriteStreamSetProperty(_stream, key._cfObject, property)
     }
+    
+    public override func propertyForKey(_ key: String) -> AnyObject? {
+        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    }
+    
 }
 
 // Discussion of this API is ongoing for its usage of AutoreleasingUnsafeMutablePointer
 #if false
```

Comment:
Can we remove this block? We'd really like to get this change merged into next snapshot release if possible. I think people are understanding that APIs may change, but right now, we're blocked from functional behavior.

Looking forward to uncommenting this and getting it all merged. 👍